packages/entity-server/src/routes/hierarchy/middleware/filter.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { Writable } from '../../../types';
+import { EntityFilter } from '../../../models';
+import { EntityFilterQuery, NestedFilterQueryFields } from '../types';
+
+const filterableFields: (keyof EntityFilterQuery)[] = [
+  'id',
+  'code',
+  'country_code',
+  'name',
+  'image_url',
+  'type',
+  'attributes_type',
+];
+const isFilterableField = (field: string): field is keyof EntityFilterQuery =>
+  (filterableFields as string[]).includes(field);
+
+const nestedFields: (keyof NestedFilterQueryFields)[] = ['attributes_type'];
+const isNestedField = (field: keyof EntityFilterQuery): field is keyof NestedFilterQueryFields =>
+  (nestedFields as (keyof EntityFilterQuery)[]).includes(field);
+
+type JsonBKey<T extends keyof NestedFilterQueryFields> = T extends `${infer Field}_${infer Key}`
```

Comment:
`JsonBKey` is a cool type that can be generalised, perhaps we could use something more generic than `NestedFilterQueryFields` and also move it to `entity-server/src/types`.

Don't worry about it if it's too inconvenient here